pkg/auth/auth.go

```diff
@@ -465,6 +465,9 @@ func (a *Authenticator) SetCSRFCookie(path string, w *http.ResponseWriter) {
 		Path:     path,
 		Secure:   a.secureCookies,
 	}
+	if a.secureCookies {
```

Comment:
The secure cookies check is actually no longer needed. This should work fine for http and https.

I'd add this to the session cookie as well.